e2e/test/scenarios/admin/performance/dashboardsAndQuestions.cy.spec.ts

```diff
@@ -0,0 +1,263 @@
+              "questions can inherit duration strategy from database",
+            strategy: sampleDurationStrategy,
+            item: sampleQuestion,
+            inheritsStrategyFrom: sampleDatabase,
+          },
+          {
+            description:
+              "questions can inherit duration strategy from instance-wide default policy",
+            strategy: sampleDurationStrategy,
+            item: sampleQuestion,
+            inheritsStrategyFrom: instanceDefault,
+          },
+          {
+            description: "dashboards can use adaptive strategy",
+            strategy: sampleAdaptiveStrategy,
+            item: sampleDashboard,
+          },
+          {
+            description:
+              "dashboards can inherit adaptive strategy from database",
+            strategy: sampleAdaptiveStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: sampleDatabase,
+          },
+          {
+            description:
+              "dashboards can inherit adaptive strategy from instance-wide default policy",
+            strategy: sampleAdaptiveStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: instanceDefault,
+          },
+          {
+            description: "dashboards can use duration strategy",
+            strategy: sampleDurationStrategy,
+            item: sampleDashboard,
+          },
+          {
+            description:
+              "dashboards can inherit duration strategy from database",
+            strategy: sampleDurationStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: sampleDatabase,
+          },
+          {
+            description:
+              "dashboards can inherit duration strategy from instance-wide default policy",
+            strategy: sampleDurationStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: instanceDefault,
+          },
+        ] as CacheTestParameters[]
+      ).forEach(
+        ({
+          strategy,
+          item,
+          inheritsStrategyFrom,
+          description,
+          it = itFunction,
+        }) => {
+          it(description, () => {
+            const { reload, visitItem } =
+              item.model === "question"
+                ? setupQuestionTest(sampleQuestion)
+                : setupDashboardTest(sampleDashboard, sampleQuestion);
+
+            const itemToConfigure = inheritsStrategyFrom ?? item;
+
+            cy.then(function () {
+              selectCacheStrategy({
+                item: itemToConfigure,
+                strategy,
+              });
+
+              visitItem();
+            });
+
+            cy.then(function () {
+              reloadUntil(resultIsCached, {
+                logMessage: "Reloading until the result is cached",
+                wrapFinalResultAs: "cachedResult",
+                wrapTimeOfFinalResultAs: "firstCachedResultSeenAt",
+                reload,
+              });
+            });
+
```

Comment:
maybe a simpler pattern here would be: 
- set db to "old value" `queryWritableDB()`
- run query to activate cache
- set db to "new value" `queryWritableDB()`
- advance clock to right before cache expires
- check that old value is still there
- advance clock to right after cache expires
- check that new value is there